### DISCUSSION TITLE
DataGrid with primitive type

### DISCUSSION THREAD

**Discussion by RealityMasque** [Megabit/Blazorise#5504] [Q&A]
really stupid question...  I have:

                    <DataGrid TItem="string" 
                              Data="@svfFeatureNames"

where svfFeatureNames is a List of string.

for some reason, the datacolumn isn't rendering each string.  I have:

                            <DataGridColumn PreventRowClick TItem="string" Caption="Feature Name" />

I'm used to providing something in the Field attribute of DataGridColumn, but I'm not sure what value should be.

As is, the number of rows is the same as in the list, but none of the actual strings get rendered.

does someone know?

thanks in advance!

**Comment by David-Moreira** [comment]
It's preferable if you wrap it in a class, as the `DataGrid` is prepared to deal with complex types (user defined classes) where you specify which `Field` it should grab the value from. 

However if you're really keen on making it work with that string list, it might work with using the `DisplayTemplate` and accessing the context. I haven't tried it before but it should work.

Since in this case it's a simpler list maybe you could look into some simpler extensions like the `ListView`.

**PS:** Also just being picky but string is not a primitive type as it is still a "complex type" / object behind the scenes. :P
